ansible_collections/arista/avd/plugins/action/validate_and_template.py

```diff
@@ -0,0 +1,168 @@
+            if not isinstance(self.dest, str):
+                raise AnsibleActionFail("The argument 'dest' must be a string")
+            schema = self._task.args["schema"]
+            if not isinstance(schema, dict):
+                raise AnsibleActionFail("The argument 'schema' must be a dict")
+        else:
+            raise AnsibleActionFail("The arguments 'template', 'dest' and 'schema' must be set")
+
+        self.conversion_mode = self._task.args.get("conversion_mode", "debug")
+        if not isinstance(self.conversion_mode, str):
+            raise AnsibleActionFail("The argument 'conversion_mode' must be a string")
+        if self.conversion_mode not in VALID_CONVERSION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{self.conversion_mode}' for the argument 'conversion_mode'. Must be one of {VALID_CONVERSION_MODES}")
+
+        self.validation_mode = self._task.args.get("validation_mode", "warning")
+        if not isinstance(self.validation_mode, str):
+            raise AnsibleActionFail("The argument 'validation_mode' must be a string")
+        if self.validation_mode not in VALID_VALIDATION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{self.validation_mode}' for the argument 'validation_mode'. Must be one of {VALID_VALIDATION_MODES}")
+
+        self.add_md_toc = self._task.args.get("add_md_toc", False)
+        if not isinstance(self.add_md_toc, bool):
+            raise AnsibleActionFail("The argument 'add_md_toc' must be a boolean")
+
+        self.md_toc_skip_lines = self._task.args.get("md_toc_skip_lines", 0)
+        if not isinstance(self.md_toc_skip_lines, int):
+            raise AnsibleActionFail("The argument 'md_toc_skip_lines' must be an integer")
+
+        # Build data from hostvars and role default vars
+        self.hostname = task_vars["inventory_hostname"]
+        self.data = self._templar.template(self._task._role.get_default_vars())
+        self.data.update(task_vars["hostvars"].get(self.hostname))
+
+        try:
+            self.avd_schema = AvdSchema(schema)
+        except Exception as e:
+            raise AnsibleActionFail("Invalid Schema supplied to the 'arista.avd.validate_and_template' plugin") from e
+
+        result_messages = []
+
+        # Perform data conversions
+        if self.conversion_mode != "disabled":
+            self.convert_data(result_messages)
+
+        # Perform validation
+        if self.validation_mode != "disabled":
+            valid = self.validate_data(result_messages)
+            if not valid and self.validation_mode == "error":
+                result["failed"] = True
+
+        if result_messages:
+            result["msg"] = " ".join(result_messages)
+
+        # Template to file
+        # Update result from Ansible "copy" operation (setting 'changed' flag accordingly)
+        result.update(self.template(task_vars))
+
+        return result
+
+    def convert_data(self, result_messages):
```

Comment:
Any reason to pass the input `result_messages` rather than returning a list in this method?

Reply:
Because this is not always called in case of disabled mode, so we would have to use extend on the calling function. This is simpler IMO.